Module6 task1

DariaEgarmina opened the pull request.

DariaEgarmina committed:
Добавляет в модель метод который собирает все необходимые данные в один объект WIP куда-то делить фотографии

DariaEgarmina committed:
Вносит изменения в осн презентер после изменений в модели

DariaEgarmina committed:
Добавляет в форму редактирования фотографии

DariaEgarmina committed:
Наследует EventEditView от AbstractStatefulView, добавляет восстановление обработчиков

DariaEgarmina committed:
Начинает связывать изменение пункта назначения и обновления картинок и описания

DariaEgarmina committed:
Связывает изменение пункта назначения и обновления картинок и описания (?)

DariaEgarmina committed:
Дорабатывает связь между названием точки маршрута и описанием и картинками

DariaEgarmina committed:
Делает так, чтобы смене дополнительных опций ранее выбранные пользователем значения не сохранялись

DariaEgarmina committed:
Изменения после обновления в редакторе ?

DariaEgarmina committed:
Добавляет и настраивает flatpicker

wonderjs commented on src/view/event-edit-form-view.js:
```diff
@@ -1,5 +1,9 @@
+import { createPhotoListTemplate } from './event-create-form-view.js';
+
+import flatpickr from 'flatpickr';
+import 'flatpickr/dist/flatpickr.min.css';
```
Хорошей практикой считается размещать импорты сторонних библиотек раньше импортов своих модулей.

wonderjs commented on src/view/event-edit-form-view.js:
```diff
@@ -137,56 +145,142 @@ const createEventEditFormTemplate = (event, destination, offer, checkedOffers) =
+  #destinationChangeHandler = (evt) => {
+    const value = evt.target.value;
+
+    evt.preventDefault(); //??нужно ли??
```
Да, нужно. Кроме того, в большинстве случаев стоит писать preventDefault в самом начало обработчика. Чтобы сразу было видно, что основное действие события будет остановлено.
Исключение: если сначала нужно сделать проверку каких-то условий и только в зависимости от них отменять поведение по умолчанию.

wonderjs commented on src/view/event-edit-form-view.js:
```diff
@@ -137,56 +145,142 @@ const createEventEditFormTemplate = (event, destination, offer, checkedOffers) =
+
+  #dueDateChangeHandler = ([userDate]) => {
+    this.updateElement({
+      dueDate: userDate,
```
Откуда взялась dueDate? Из демки по учебному проекту? =) 
У нас событие имеет dateFrom и dateTo. На каждую нужен свой обработчик, потому что будем обновлять разные ключи.

wonderjs commented on src/view/event-edit-form-view.js:
```diff
@@ -137,56 +145,142 @@ const createEventEditFormTemplate = (event, destination, offer, checkedOffers) =
+      {
+        dateFormat: 'd/m/y H:i',
+        enableTime: true,
+        defaultDate: this._state.dueDate,
```
Вместо dueDate должно быть dateFrom / dateTo.
Кроме дефолтной даты, ещё хорошо бы задать минимальную для dateTo (чтобы дата окончания не была раньше даты начала) и максимальную для dateFrom (чтобы дата начала не была позже даты окончания).

wonderjs reviewed (changes requested):
Форму создания тоже оживляем и вешаем обработчики. Либо колдуем, чтобы использовался один и тот же компонент и для формы редактирования и для формы создания, чтобы не дублировать обработчики. Его тогда можно назвать просто EventFormView. 
![photo_2025-05-14 11 55 21](https://github.com/user-attachments/assets/c09f150a-ffe6-4a83-91a8-b6aefdf15db6)

DariaEgarmina committed:
Вносит правки в реализацию flatpicker

DariaEgarmina committed:
Создает view для кнопки New event и отрисовывет её

DariaEgarmina committed:
Делает заготовку обработчика события клика по кнопке New event, WIP

DariaEgarmina committed:
Делает презентер для фильтров и кнопки в хедере WIP

Пока добавляет в этот презентер только отрисовку кнопки, с фильтрами разберется позже

DariaEgarmina committed:
Реализует передачу обработчика нажатия на кнопку через презентер WIP

DariaEgarmina committed:
Повторяет поведение контролов и валидации для компонента формы добавления новой точки маршрута WIP

DariaEgarmina committed:
Исправляет ошибку при вводе пункта назначения, которого нет в списке WIP
